src/borg/archiver/extract_cmd.py

```diff
@@ -56,33 +56,51 @@ def do_extract(self, args, repository, manifest, archive):
         else:
             pi = None
 
-        for item in archive.iter_items(filter):
-            archive.preload_item_chunks(item, optimize_hardlinks=True)
+        for item in archive.iter_items():
             orig_path = item.path
-            if strip_components:
-                item.path = os.sep.join(orig_path.split(os.sep)[strip_components:])
-            if not args.dry_run:
-                while dirs and not item.path.startswith(dirs[-1].path):
-                    dir_item = dirs.pop(-1)
-                    try:
-                        archive.extract_item(dir_item, stdout=stdout)
-                    except BackupError as e:
-                        self.print_warning_instance(BackupWarning(remove_surrogates(dir_item.path), e))
+            components = orig_path.split(os.sep)
+            stripped_path = os.sep.join(components[strip_components:])
+
+            if not stripped_path:
+                continue
+            item.path = stripped_path
```

Comment:
the original code only did that "if strip_components".

Reply:
also, you could stay closer to the original code (like: "don't fix what ain't broken") and just add the continue if item.path ends up being empty.

Reply:
Got it. The only issue is, that just adding continue resulted in test failures due to a `ValueError` being raised. `ValueError: path '' is not sanitized`. To resolve this, I updated the code to ensure that `item.path` is only set if the resulting path is not empty.